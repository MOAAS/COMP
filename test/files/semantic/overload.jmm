import Quicksort;
import Quicksort.foo(int);

class Simple extends Quicksort {
    public static void main(String[] arg){
        Simple simple;
        simple = new Simple();
        simple.foo(1);
        simple.foo(1, 1);
        simple.foo(true, 1);
        simple.foo(1, true);
        simple.foo(true);
        simple.foo(simple);


    }

    public int foo(boolean a){
       this.foo(10);
       this.foo(true);
       return 1;
    }

    public int foo(Simple a){
       return 1;
    }

    public int foo(boolean a, int b){
       return 1;
    }
}