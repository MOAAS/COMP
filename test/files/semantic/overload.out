Error: Class "Simple" does not contain method "foo(int,int)" at line 9, column 9.
Error: Class "Simple" does not contain method "foo(int,boolean)" at line 11, column 9.
